src/Components/SmartComponents/Modals/OsExposureReportModal/OsExposureReportModal.js

```diff
@@ -0,0 +1,234 @@
+import {
+    Modal,
+    Button,
+    Form,
+    FormGroup,
+    TextInput,
+    Flex,
+    FlexItem,
+    Checkbox,
+    TextArea
+} from '@patternfly/react-core';
+import LifecycleSelect from './LifecycleSelect';
+import isEqual from 'lodash/isEqual';
+import { CUSTOM_REPORT_FILTER_DATA_BY_LABEL, getCveReportFilters } from '../../../../Helpers/constants';
+
+const OsExposureReportModal = ({
+    isOsExposureModalOpen,
+    setOsExposureModalOpen,
+    osExposureData
+}) => {
+    const [reportTitle, setReportTitle] = useState('');
+    const [exposureMap, setExposureMap] = useState({});
+    const [selectedOsVersions, setSelectedOsVersions] = useState([]);
+    const [filterData, setFilterData] = useState({
+        impact: [],
+        advisory_available: ['true']
+    });
+    const [userNote, handleUserNoteChange] = useState('');
+    const CVE_REPORT_FILTERS = getCveReportFilters(false, true);
+    const ACTIVE_FILTERS = [
+        'impact',
+        'advisory_available'
+    ];
+
+    useEffect(() => {
+        if (osExposureData && osExposureData.length) {
+            const exposureMapCopy = { ...exposureMap };
+            osExposureData.sort((a, b) => {
+                if (a.major < b.major) {return 1;}
+
+                if (a.major > b.major) {return -1;}
+
+                if (a.minor < b.minor) {return 1;}
+
+                if (a.minor > b.minor) {return -1;}
+            }).forEach((os) => {
+                const majorVer = `${os.name} ${os.major}`;
+                const minorVer = `${os.major}.${os.minor}`;
+                const lifecycle = `${os.lifecycle_phase}`;
+
+                if (!exposureMapCopy[majorVer]) {
+                    exposureMapCopy[majorVer] = {};
+                }
+
+                if (!exposureMapCopy[majorVer][minorVer]) {
+                    exposureMapCopy[majorVer][minorVer] = {};
+                }
+
+                exposureMapCopy[majorVer][minorVer][lifecycle] = os;
+            });
+
+            setExposureMap(exposureMapCopy);
+        }
+    }, [osExposureData]);
+
+    const isSameOsMinor = (osMinor) => {
+        let isSelected = false;
+        selectedOsVersions.forEach((os) => {
+            if (isEqual(os, osMinor)) {
+                isSelected = true;
+                return;
+            }
+        });
+
+        return isSelected;
+    };
+
+    const toggleArrayItem = (item) => {
+        let selectedOsVersionsCopy = selectedOsVersions.filter((os) => !isEqual(os, item));
+        if (selectedOsVersionsCopy.length === selectedOsVersions.length) {
+            selectedOsVersionsCopy.push(item);
+        }
+
+        selectedOsVersionsCopy.sort((a, b) => {
+            if (Object.keys(a)[0] < Object.keys(b)[0]) {
+                return 1;
+            } else if (Object.keys(a)[0] > Object.keys(b)[0]) {
+                return -1;
+            }
+
+            return 0;
+        });
+
+        setSelectedOsVersions(selectedOsVersionsCopy);
+    };
```

Comment:
As per our discussion in Slack, I believe this could be simplified to:

```js
const toggleArrayItem = (item) => {
  const unsorted = _.xorWith(selectedOsVersions, [item], isEqual);
  const sorted = _.sortBy(unsorted, (obj) => Object.keys(obj)[0]); // There might be a need to append ".reverse()" at the end here
  setSelectedOsVersions(sorted);
};
```

Reply:
I definitely agree with you that one-liners can be dangerous and could cause the code to be hard to read, but each of these lines contains one or two function calls with just a few arguments and iś IMO quite easy to parse by eye.